### ISSUE TITLE
Is this LoRA implementation correct?

### ISSUE THREAD

**Issue by AayushManShrestha** [m87-labs/moondream#192]
I was trying to fine-tune Moondream2 by using LoRA. But, I got weird loss curves.
Here is the link to the code: [LoRA-finetune](https://colab.research.google.com/drive/1cNSme3Tc3vynxUmov-JN3uEzbU3mbG9W?usp=sharing)
Here is the loss curve when doing fine-tuning normally: 
![2UCqaLX](https://github.com/user-attachments/assets/04c60e0f-f1ad-4127-b17c-84c22d07500a)

Here is the loss curve when using LoRA for fine-tuning
![2UCqlBn](https://github.com/user-attachments/assets/5a35a514-501e-4f44-a008-a13f3ddce1bf)
